src/components/ha-push-notifications-toggle.js

```diff
@@ -88,8 +91,12 @@ class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
         browserName = "chrome";
       }
 
-      const name = prompt("What should this device be called ?");
-      if (name == null) {
+      const name = await showPromptDialog(this, {
+        title: "Name",
+        text: "What should this device be called?",
+      });
+
+      if (!name) {
```

Comment:
Check for `null` as that is the value as you cancel. Otherwise, it is empty but not canceled.